### DISCUSSION TITLE
Coil Remains Engaged After MPF <--> OPP Comms Fail

### DISCUSSION THREAD

**Discussion by founfabug** [missionpinball/missionpinball#91] [Q&A]
I recently had a high resistance ground path develop on my 3 board OPP setup. This resulted in comms failure between OPP board serial No 11 and MPF. MPF terminated with a communications read error. A non-autofire coil on board serial No 10 was in mid pulse at the time, and remained energised after MPF terminated. This was repeatable while I tracked down the high resistance ground path. Is this expected behaivour?  

I've designed a watchdog and power supervisory circuit for my coil power, which monitors a slow square wave produced from a show in config.yaml that starts with mode_attract_started. Are there better solutions to coil-lock-on problems?

Thank you.

**Comment by cobra18t** [comment]
Could you share your coil config? Was it mid-pulse or mid-enable? OPP controls the pulse length, but MPF controls enable/disable commands.

What version of OPP firmware are you using? Do you have a log?

**Reply by cobra18t** [reply]
Also, was the firmware still running after the lost comms? Was the blue LED on the processor still flashing slowly?

**Comment by founfabug** [comment]
At the time I was testing **enable** with the coil concerned, so you have solved my mystery for me, thank you. My STM units lack a useful blue LED, they are based on the 2016 design with PC13 LED and power LED only. However the coil was audibly PWMing, rather than simply on.  The firmware is latest - 2.4.0.0.

From this I assume it is wise to avoid enable/disable with coils that can't run all day with whatever duty cycle is set for them. 

I read up a little about the IWDG in the STM, and the OPP firmware uses it ("while petting the watchdog"), but I lack the ability to understand what is done with it. Do you know if it is configured to raise NRST as I have seen done elsewhere?

Thank you again.

**Comment by cobra18t** [comment]
OPP is designed to run without a host in a whitewood mode, so it will still drive/pulse coils if MPF leaves them enabled or does not disable autofire devices.

That depth of OPP question will have to be directed to OPP himself. I don't know.

**Reply by founfabug** [reply]
Thank you again!
